service/src/test/java/uk/gov/hmcts/reform/fpl/service/document/DocumentGeneratorTest.java

```diff
@@ -0,0 +1,84 @@
+import uk.gov.hmcts.reform.fpl.enums.OrderStatus;
+import uk.gov.hmcts.reform.fpl.enums.docmosis.RenderFormat;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+import uk.gov.hmcts.reform.fpl.model.common.DocmosisDocument;
+import uk.gov.hmcts.reform.fpl.model.configuration.Language;
+import uk.gov.hmcts.reform.fpl.model.order.Order;
+import uk.gov.hmcts.reform.fpl.service.docmosis.DocmosisDocumentGeneratorService;
+import uk.gov.hmcts.reform.fpl.service.orders.docmosis.DocmosisParameters;
+import uk.gov.hmcts.reform.fpl.service.orders.generator.DocmosisCommonElementDecorator;
+import uk.gov.hmcts.reform.fpl.service.orders.generator.DocmosisParameterGenerator;
+
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static uk.gov.hmcts.reform.fpl.enums.OrderStatus.SEALED;
+
+@ExtendWith(MockitoExtension.class)
+class DocumentGeneratorTest {
+
+    private static final Map<String, Object> TEMPLATE_DATA = Map.of("key", "value");
+    private static final DocmosisTemplates DOCMOSIS_TEMPLATE = DocmosisTemplates.ORDER_V2;
+    private static final DocmosisDocument DOCMOSIS_DOCUMENT = mock(DocmosisDocument.class);
+    private static final DocmosisParameters DOCMOSIS_PARAMETERS = mock(DocmosisParameters.class);
+    private static final CaseData CASE_DATA = mock(CaseData.class);
+    private final DocmosisParameterGenerator parameterGenerator = mock(DocmosisParameterGenerator.class);
+    private static final Order ORDER = Order.C32A_CARE_ORDER;
+    private static final RenderFormat FORMAT = RenderFormat.PDF;
+    private static final OrderStatus STATUS = SEALED;
+
+    @Mock
+    private DocmosisDocumentGeneratorService docmosisRenderer;
+
+    @Mock
+    private DocmosisCommonElementDecorator decorator;
+
+    @Mock
+    private ObjectMapper objectMapper;
+
+    @InjectMocks
+    private DocumentGenerator underTest;
+
+    @BeforeEach
+    void setUp() {
+        when(decorator.decorate(DOCMOSIS_PARAMETERS, CASE_DATA, STATUS, ORDER)).thenReturn(DOCMOSIS_PARAMETERS);
+        when(objectMapper.convertValue(
+            eq(DOCMOSIS_PARAMETERS),
+            Mockito.<TypeReference<Map<String, Object>>>any())).thenReturn(TEMPLATE_DATA);
+        when(docmosisRenderer.generateDocmosisDocument(TEMPLATE_DATA, DOCMOSIS_TEMPLATE, FORMAT, Language.ENGLISH))
+            .thenReturn(DOCMOSIS_DOCUMENT);
+        when(parameterGenerator.accept()).thenReturn(ORDER);
+        when(parameterGenerator.template()).thenReturn(DOCMOSIS_TEMPLATE);
+        when(parameterGenerator.generate(CASE_DATA)).thenReturn(DOCMOSIS_PARAMETERS);
+    }
+
+    @Test
+    void shouldGenerateDocument() {
+        DocmosisDocument generatedDocument = underTest.generateDocument(CASE_DATA, parameterGenerator, FORMAT, STATUS);
+
+        assertThat(generatedDocument).isEqualTo(DOCMOSIS_DOCUMENT);
+    }
+
+    @Test
```

Comment:
These tests are not perfect  right, returns same doc regardless of what params are passed. From test it looks that format is always PDF and status is SEALED

Reply:
That's because, as far as the tested class is concerned, these two parameters make no difference. It's just making sure these are relayed to other classes.